File: form-generator/formGenerator.py
import json, os
from menu import menu

cName = "config.json" # Configuration file name

def text(*text: str):
    # Shows a text on the question
    print(*text)

def label(question: str, *args):
    return input(question + "\n")

def choice(question: str, *options):
    r = menu(question, *options)
    return options[r - 1]

cvTable = {
    "text": text,
    "choice": choice,
    "label": label
}

def run(form: dict, length: int = 40):
    responses = []
    for i, k in enumerate(form["func"]):
        os.system('cls' if os.name == 'nt' else 'clear')

        print(f"Question {i + 1}")
        print("=" * length)

        responses.append(cvTable[k](*form["args"][i]))
        print("=" * length)

        input("Press Enter to go to the next question...")

    return responses

def main():
    if os.path.exists(cName):
        # load file
        with open(cName, "r") as f:
            c = json.loads(f.read())

        r = run(c)
        print(r)

    else:
        # create file
        with open(cName, "w") as f:
            f.write(json.dumps({}))

        print("file created")

if __name__ == "__main__":
    main()

File: form-generator/formGenerator.php
<!DOCTYPE html>
<html>
    <title>
        Form generator
    </title>
    <body>
    <?php
        $cName = "config.json";
        $form = json_decode(file_get_contents($cName), true);

        echo "<h1>$form[title]</h1>\n";

        function text($text) {
            echo "<p>$text</p>\n";
        }

        function label($question, $id, $name = null) {

            if ($name === null) {
                $name = $id;
            }

            echo "<label for=$id>$question</label><br>\n";
            echo "<input type=text id=$id name=$name>\n";
        }

        function radio($question, $name, $options) {
            echo "<p>$question</p>";
            foreach($options as $op) {
                echo "<input type=radio id=" . $op["id"] . " name=$name  value=" . $op["value"] . ">";
                echo "<label for=$name>" . $op["content"] . "</label>";
            }
        }

        $cont = 0;
        echo "<form>";
        foreach($form["func"] as $f) {
            echo "<h2>Question" . $cont + 1  . "</h2>\n";
            call_user_func_array($f, $form["args"][$cont]);
            $cont++;
        }
        echo "</form>";
    ?>
    </body>
</html>